### PR TITLE
Testdata: support null values in manual entry and return all data

### PR DESCRIPTION
**Pull request by ryantxu**
This makes two modifications to the manual entry test data scenario
1. it does not crash with null values
2. it returns all the data regardless of the query time range

not crashing is obvious, (2) is a breaking change, but it seems like this purpose of manual entry testdata is to have it return data you enter manually.  We need to return results with explicit timestamps and null values *outside* the requested query range.

Given that the UI is completely unusable, i doubt this change will affect anyone
image

### CODE REVIEW

File: pkg/tsdb/testdatasource/scenarios.go [hunk 1]
```diff
@@ -285,42 +285,40 @@ func (p *testDataPlugin) handleManualEntryScenario(ctx context.Context, req *bac
...
 		points := model.Get("points").MustArray()
 
 		frame := newSeriesForQuery(q, model, 0)
-		startTime := q.TimeRange.From.UnixNano() / int64(time.Millisecond)
-		endTime := q.TimeRange.To.UnixNano() / int64(time.Millisecond)
 
-		timeVec := make([]*time.Time, 0)
-		floatVec := make([]*float64, 0)
+		timeField := data.NewFieldFromFieldType(data.FieldTypeTime, 0)
+		valueField := data.NewFieldFromFieldType(data.FieldTypeNullableFloat64, 0)
```
Comment (ryantxu) [inline pkg/tsdb/testdatasource/scenarios.go:295]: this change it not really necessary... but worth noting that creating fields from a slice will copy them an extra time vs this method that appends directly to the final vaue

File: pkg/tsdb/testdatasource/scenarios.go [hunk 2]
```diff
@@ -285,42 +285,40 @@ func (p *testDataPlugin) handleManualEntryScenario(ctx context.Context, req *bac
...
+		timeField.Name = "time"
+		valueField.Name = "value"
 
 		for _, val := range points {
 			pointValues := val.([]interface{})
 
-			var value float64
+			var value *float64 = nil
```
Comment (aknuds1) [inline pkg/tsdb/testdatasource/scenarios.go:302]: It's superfluous/more difficult to read to initialize a zero value explicitly.
```suggestion
			var value *float64
```

File: pkg/tsdb/testdatasource/scenarios_test.go [hunk 3]
```diff
@@ -185,6 +185,63 @@ func TestTestdataScenarios(t *testing.T) {
...
+
+	t.Run("manual entry ", func(t *testing.T) {
+		t.Run("should support nulls and return all data", func(t *testing.T) {
+			timeRange := plugins.DataTimeRange{From: "5m", To: "now", Now: time.Now()}
+
+			// body := strings.NewReader(
+
+			// model := simplejson.NewFromReader(body)
+			// modelBytes, err := model.MarshalJSON()
+			// require.NoError(t, err)
```
Comment (aknuds1) [inline pkg/tsdb/testdatasource/scenarios_test.go:197]: Can you remove this dead code?

**Review by aknuds1 — APPROVED** [review]
LGTM, with a couple of nits.

**Pull request merged**